nativelib/src/main/resources/gc/immix/ImmixGC.c

```diff
@@ -13,8 +14,56 @@
 
 void scalanative_collect();
 
+/*
+ Accepts number of bytes or number with a suffix letter for indicating the units.
+ k or K for kilobytes(1024 bytes), m or M for megabytes and g or G for gigabytes.
+*/
+size_t parseSizeStr(char * str) {
+    int length = strlen(str);
+    size_t size;
+    sscanf(str, "%zu", &size);
+    char possibleSuffix = str[length - 1];
+    switch (possibleSuffix) {
+        case 'k':
+        case 'K':
+            size *= 1024;
+            break;
+        case 'm':
+        case 'M':
+            size *= 1024 * 1024;
+            break;
+        case 'g':
+        case 'G':
+            size *= 1024 * 1024 * 1024;
+    }
+    return size;
+}
+
 NOINLINE void scalanative_init() {
-    Heap_Init(&heap, INITIAL_SMALL_HEAP_SIZE, INITIAL_LARGE_HEAP_SIZE);
+    char *minHeapSizeStr = getenv("SCALANATIVE_MIN_HEAP_SIZE");
+    size_t minHeapSize;
+    if (minHeapSizeStr != NULL) {
+        minHeapSize = parseSizeStr(minHeapSizeStr);
+    } else {
+        minHeapSize = DEFAULT_MIN_SMALL_HEAP_SIZE;
+    }
+
+    if (minHeapSize < 2 * BLOCK_TOTAL_SIZE) {
```

Comment:
Why 2 blocks? This seems to be *really* low, even it works now. Scala Native is not aimed at super lower memory devices with <1M of memory. Lets set a lower limit at 1M.